internal/controller/storageclient_controller.go

```diff
@@ -55,54 +64,143 @@ const (
 	GetStorageConfig      = "GetStorageConfig"
 	AcknowledgeOnboarding = "AcknowledgeOnboarding"
 
-	storageClientNameLabel             = "ocs.openshift.io/storageclient.name"
-	storageClientFinalizer             = "storageclient.ocs.openshift.io"
-	storageClaimProcessedAnnotationKey = "ocs.openshift.io/storageclaim.processed"
-	storageClientDefaultAnnotationKey  = "ocs.openshift.io/storageclient.default"
+	storageClientNameLabel = "ocs.openshift.io/storageclient.name"
+	storageClientFinalizer = "storageclient.ocs.openshift.io"
 
 	// indexes for caching
-	ownerIndexName = "index:ownerUID"
+	pvClusterIDIndexName   = "index:persistentVolumeClusterID"
+	vscClusterIDIndexName  = "index:volumeSnapshotContentCSIDriver"
+	vgscClusterIDIndexName = "index:volumeGroupSnapshotContentCSIDriver"
+	ownerUIDIndexName      = "index:ownerUID"
+
+	VolumeGroupSnapshotClassCrdName = "volumegroupsnapshotclasses.groupsnapshot.storage.k8s.io"
 
 	csvPrefix = "ocs-client-operator"
 )
 
 // StorageClientReconciler reconciles a StorageClient object
 type StorageClientReconciler struct {
-	ctx context.Context
 	client.Client
-	Log           klog.Logger
-	Scheme        *runtime.Scheme
-	recorder      *utils.EventReporter
-	storageClient *v1alpha1.StorageClient
-
+	Scheme            *runtime.Scheme
 	OperatorNamespace string
+	AvailableCrds     map[string]bool
+
+	log               klog.Logger
+	ctx               context.Context
+	recorder          *utils.EventReporter
+	storageClient     *v1alpha1.StorageClient
+	storageClientHash string
```

Comment:
You should not store request specific information on a reconciler object that is shared across all reconciles

Reply:
pinged for ack to make changes offline.

Reply:
will be worked in a different PR.